File: hub/src/light/effect/color_effects/single_color_effect.py
from light.effect.partial_effect import PartialEffect
from light.effect.resolved_effect import ResolvedEffect
from light.effect.effect_priority import EffectPriority

class SingleColorEffect(PartialEffect):
    def __init__(self, rgbw, dt=0):
        super().__init__(dt)
        self.rgbw = rgbw
    
    def getEffect(self, t):
        return ResolvedEffect.createRgbw(self.rgbw)

    def isTemporal(self):
        return False

    def isComplete(self, t):
        return False

File: hub/drumlite-js/src/light/effect/color/SingleColorEffect.ts
import PartialEffect, { EffectParameters, EffectParameter } from "../PartialEffect"
import RGB from "../../RGB"
import ResolvedEffect from "../../../effect/ResolvedEffect";

export class SingleColorEffectParams extends EffectParameters {
    effectName = "Single Color";
    typeName = "Color";

    constructor(rgb: RGB = new RGB()) {
        super(0);
        this.params.rgb = new EffectParameter<RGB>("Color", rgb, {type: "rgb"});
    }
}

export default class SingleColorEffect extends PartialEffect<SingleColorEffectParams> {
    constructor(params: SingleColorEffectParams, dt: number = 0) {
        super(params, dt);
    }
    
    public getEffect(t: number) {
        return [[ResolvedEffect.createRgb(this.params.params.rgb.val)]];
    }

    public isTemporal() {
        return false;
    }

    public isComplete(t: number) {
        return false;
    }
}